src/models/document/document-content.ts

```diff
@@ -667,11 +667,16 @@ export const DocumentContentModel = types
           if (row?.tiles.length === rowTiles.length) {
             if ((rowDropLocation === "left") || (rowDropLocation === "right")) {
               // entire row is being merged with an existing row
-              self.mergeRow(row, rowInfo);
+              const dstRow = rowInfo.rowDropIndex ? self.getRowByIndex(rowInfo.rowDropIndex) : undefined;
```

Comment:
Note that while this looks like an opportunity to use nullish coalescing, TypeScript throws an error that `rowInfo.rowDropIndex` could be undefined when passed to `getRowByIndex()` which seems like a TypeScript bug since the whole point of nullish coalescing is that the rest of the expression won't be evaluated if it's null/undefined.